### ISSUE TITLE
Embendings 3d models

### ISSUE THREAD

**Issue by Danilka234** [Ayumu-J-S/Point-JEPA#1]
Good day. Thank you for your efforts! I wanted to clarify, usually such neural networks are demonstrated with such datasets as - modelnet, shapenet and other. But I need as many 3D models as possible. I found such a dataset - ABC. There are 1 million models, but no classes.
I need this neural network just to extract embendings to fill the vector database for subsequent comparison. Do I understand correctly that I can do this using this neural network?
Sorry for the stupid questions, but I will be grateful if someone helps me. :)

**Comment by Ayumu-J-S** [comment]
Thanks for your interest in our work:) That should be possible. Ideally, you'd want a dataset with similar geometric features to ShapeNet for optimal performance. However, since the goal is to extract embeddings for a vector database rather than classification, you should still be able to use this neural network effectively.

If your dataset (ABC) contains a diverse set of 3D models without predefined classes, the network can still generate meaningful embeddings, provided that instances with different features (think of cars vs. cats for example). From there, you should be able to cluster the embeddings or just visualize the input instances in each cluster to see if they turned out fine.

If this doesn’t yield good results, an alternative is to fine-tune the pretrained network on your dataset. You'd just need to define a new dataset class and train it accordingly.

For implementation details on how to extract embeddings, [this piece of code](https://github.com/Ayumu-J-S/Point-JEPA/blob/93d382c6fe926846b3ad7b59003ba862a340f37c/pointjepa/eval/embedding_cluster_vis.py#L42C1-L44C1) might help.


Let me know if you need further clarification!

**Comment by Danilka234** [comment]
Good afternoon Thanks for your answer!
The fact is that classes are absent in [ABC Dataset](https://archive.nyu.edu/handle/2451/43778). I previously examined with Point -Bert by exposing one fictitious class - Objects and drove off all models as one class. Then I extracted embedding and visualized. The result was certainly not very + problem with Pre-Train. The repository is already abandoned there and no one answers. I saw your work, decided to try it, at the same time and clarify whether I am moving in the right direction.
Does it make sense to train a variety of CAD models with one class or still need to distribute models by classes?
P.S. I am very glad to your answer, otherwise in such projects from one year and no longer answer questions! :)

**Comment by Danilka234** [comment]
Oh, I'm sorry, you have already given directions.
It turns out that if I need to go further to Fine -Tuning, I will need to make exactly the same class to an example - Objects. I understand correctly?

**Comment by Ayumu-J-S** [comment]
Sorry for the late reply, Yes; you will want to create a dataset class like [this one](https://github.com/Ayumu-J-S/Point-JEPA/blob/main/pointjepa/datasets/ShapeNet55.py) that returns the point cloud and the label. We are actually not using the label in the point-jepa training script so you can always return None for the label for your dataset

**Comment by Danilka234** [comment]
Thank you very much for your work and feedback! I have made progress in this direction, I close the issue.
